api/src/main/java/io/strimzi/api/kafka/model/kafka/cruisecontrol/CruiseControlSpec.java

```diff
@@ -109,6 +113,29 @@ public void setConfig(Map<String, Object> config) {
         this.config = config;
     }
 
+    @Description("Configuration of the Cruise Control REST API users.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public ApiUsers getApiUsers() {
+        if (this.apiUsers != null) {
+            checkApiUsersConfig(this.apiUsers);
```

Comment:
I would not import the method but using the `ApiUsers.checkApiUsersConfig` syntax.
Furthermore, the method is already checking that provided `apiUsers` is not null, why are you testing it again here? (line 119). I guess having `ApiUsers.checkApiUsersConfig` to return false, would mean this method returning null.

Reply:
Actually I am not sure we need the following logic at all. Shouldn't it come from the CRD generator (builders and fluent classes) so that when you have a `apiUsers` field in the CC CR, it's automatically loaded into the member field here? Looking at the `Storage` field in the `KafkaClusterSpec`. There isn't something like this method to fill the `Storage` field with the ephemeral or jbod type. Or am I missing something @scholzj ?

Reply:
> Looking at the Storage field in the KafkaClusterSpec. There isn't something like this method to fill the Storage field with the ephemeral or jbod type. 

I think you are right, let me play with this a little bit more

Reply:
Sorry, I missed this mention. But there should not be any logic like this in the `api` module. At least we do not have it anywhere else.